File: bookmarker-java/src/main/java/com/sivalabs/bookmarker/domain/repository/TagRepository.java
package com.sivalabs.bookmarker.domain.repository;

import com.sivalabs.bookmarker.domain.entity.Tag;
import org.springframework.data.jpa.repository.JpaRepository;
import java.util.Optional;

public interface TagRepository extends JpaRepository<Tag, Long> {
    Optional<Tag> findByName(String tag);
}

File: backend/src/main/kotlin/com/sivalabs/bookmarker/domain/repository/TagRepository.kt
package com.sivalabs.bookmarker.domain.repository

import com.sivalabs.bookmarker.domain.entity.Tag
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface TagRepository : JpaRepository<Tag, Long> {
    fun findByName(tag: String): Optional<Tag>
}
